tools/installpsh-debian.sh

```diff
@@ -0,0 +1,167 @@
+thisinstallerdistro=debian
+repobased=true
+gitscriptname="installpsh-debian.psh"
+
+echo ; 
+echo "*** PowerShell Core Development Environment Installer $VERSION for $thisinstallerdistro"
+echo "***    Current PowerShell Core Version: $currentpshversion"
+echo "***    Original script is at: $gitreposcriptroot/$gitscriptname"
+echo
+echo "*** Arguments used: $*"
+
+#Verify The Installer Choice (for direct runs of this script)
+lowercase(){
+    echo "$1" | sed "y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/"
+}
+
+# Let's quit on interrupt of subcommands
+trap '
+  trap - INT # restore default INT handler
+  echo "Interrupted"
+  kill -s INT "$$"
+' INT
+
+if [ "{$OS}" == "windowsnt" ]; then
+    OS=windows
+    DistroBasedOn=windows
+elif [ "{$OS}" == "darwin" ]; then
+    OS=osx
+    DistroBasedOn=osx
+else
+    OS=`uname`
+    if [ "${OS}" == "SunOS" ] ; then
+        OS=solaris
+        ARCH=`uname -p`
+        OSSTR="${OS} ${REV}(${ARCH} `uname -v`)"
+        DistroBasedOn=sunos
+    elif [ "${OS}" == "AIX" ] ; then
+        OSSTR="${OS} `oslevel` (`oslevel -r`)"
+        DistroBasedOn=aix
+    elif [ "${OS}" == "Linux" ] ; then
+        if [ -f /etc/redhat-release ] ; then
+            DistroBasedOn='redhat'
+        elif [ -f /etc/SuSE-release ] ; then
+            DistroBasedOn='suse'
+        elif [ -f /etc/mandrake-release ] ; then
+            DistroBasedOn='mandrake'
+        elif [ -f /etc/debian_version ] ; then
+            DistroBasedOn='debian'
+        fi
+        if [ -f /etc/UnitedLinux-release ] ; then
+            DIST="${DIST}[`cat /etc/UnitedLinux-release | tr "\n" ' ' | sed s/VERSION.*//`]"
+            DistroBasedOn=unitedlinux
+        fi
+        OS=`lowercase $OS`
+        DistroBasedOn=`lowercase $DistroBasedOn`
+    fi
+fi
+
+if [ "$DistroBasedOn" != "$thisinstallerdistro" ]; then
+  echo "*** This installer is only for $thisinstallerdistro and you are running $DistroBasedOn, please run \"$gitreporoot\install-powershell.sh\" to see if your distro is supported AND to auto-select the appropriate installer if it is."
+  exit 0
+fi
+
+## Check requirements and prerequisites
+
+#Only do SUDO if we are not root
+SUDO=''
+if (( $EUID != 0 )); then
+    SUDO='sudo'
+fi
+
+#Check that sudo is available
+if [[ "$SUDO" -eq "sudo" ]]; then
+
+    $SUDO -v
+    if [ $? -ne 0 ]; then
+      echo "ERROR: You must either be root or be able to use sudo" >&2
+      exit 5
+    fi
+fi
+
+#Collect any variation details if required for this distro
+REV=`cat /etc/lsb-release | grep '^DISTRIB_RELEASE' | awk -F=  '{ print $2 }'`
+#END Collect any variation details if required for this distro
+
+#If there are known incompatible versions of this distro, put the test, message and script exit here:
+
+#END Verify The Installer Choice
+
+##END Check requirements and prerequisites
+
+echo
+echo "*** Installing PowerShell Core for $DistroBasedOn..."
+if ! hash curl 2>/dev/null; then
```

Comment:
In my googling on it seemed that wget is pretty universal - this was also born out by the original download.sh.  But I like the idea, because if someone replicates the repo or copies the script set manually - that is a scenario where I can't assume wget was used to pull the script in the first place.